tests/page-objects/forms/generic-form.po.js

```diff
@@ -109,20 +124,23 @@ module.exports = {
     const submitButton = element(by.css('.btn.submit.btn-primary'));
     await helper.waitElementToBeClickable(submitButton);
     await submitButton.click();
-    await helper.waitElementToBeVisible(element(by.css('div#reports-content')));
+    await helper.waitElementToPresentNative(element(by.css('div#reports-content')));
     const details = element(by.css('div.details'));
     await helper.waitUntilReadyNative(details);
     expect(await details.isPresent()).toBeTruthy();
   },
 
-  validateReport: () => {
-    const reportValidBtn = element(by.css('[ng-include*="verify-valid"]'));
-    helper.waitElementToBeClickable(reportValidBtn);
-    reportValidBtn.click();
+  validateReportNative: async () => {
+    const reviewButton = element.all(by.css('.mm-icon-inverse')).get(0);
+    await helper.waitUntilReadyNative(reviewButton);
+    await reviewButton.click();
+    const reportValidBtn = element(by.css('.verify-valid'));
+    await helper.waitElementToBeClickable(reportValidBtn);
+    await reportValidBtn.click();
     const reportValidIcon = element(by.css('.detail>.status>.verified'));
-    helper.waitUntilReady(reportValidIcon);
+    await helper.waitUntilReadyNative(reportValidIcon);
     const reportValidMessage = element(by.css('.verify-valid>span:last-of-type'));
-    expect(reportValidMessage.getText()).toEqual('Correct');
+    expect(await reportValidMessage.getText()).toEqual('Correct');
```

Comment:
To test an additional feature, we can change this to:
```suggestion
    const reportValidMessage = element(by.css('.verify-valid.active'));
    expect(await reportValidMessage.getText()).toEqual('Correct');
```
This way we're checking that the review UI also "activates" the valid icon.

Reply:
Nice! I have made the change.